compiler/generator_rust.c

```diff
@@ -939,6 +945,7 @@ static void generate_dollar(struct generator * g, struct node * p) {
               "~Menv.limit = env.current.len();~N", p);
     generate(g, p->left);
     if (!g->unreachable) {
+        g->V[0] = p->name;
         /* Update string variable. */
         w(g, "~M~V0 = env.current.clone().into_owned();~N");
```

Comment:
Hmm, so does this make a copy of the contents of the variable and then copy it back afterwards?

If so, that's different to the C implementation because during `$s` `s` will maintain its old value (and only be updated at the end) - for example, consider:

```
externals ( stem )
strings ( s t )
define stem as (
  $s = 'foo'
  $s ( = 'bar' $t = s )
)
```

With the C backend, both `s` and `t` will be set to `bar`; if the value of `s` is only updated at the end of the `$` section then `t` would instead be set to `foo`.